tests/engine/runner/test_dask.py

```diff
@@ -0,0 +1,334 @@
+        "subtract_2": SchemaNode(
+            needs={"i": "add"},
+            uses=SubstractByX,
+            fn="subtract_x",
+            constructor_name="create",
+            config={"x": 2},
+            eager=eager,
+            is_target=True,
+        ),
+    }
+
+    execution_context = ExecutionContext(graph_schema=graph_schema, model_id="1")
+
+    runner = DaskGraphRunner(
+        graph_schema=graph_schema, execution_context=execution_context
+    )
+    results = runner.run(inputs={"first_input": 3, "second_input": 4})
+    assert results["subtract_2"] == 5
+
+
+@pytest.mark.parametrize("eager", [True, False])
+def test_target_override(eager: bool):
+    graph_schema: GraphSchema = {
+        "add": SchemaNode(
+            needs={"i1": "first_input", "i2": "second_input"},
+            uses=AddInputs,
+            fn="add",
+            constructor_name="create",
+            config={},
+            eager=eager,
+        ),
+        "subtract_2": SchemaNode(
+            needs={"i": "add"},
+            uses=SubstractByX,
+            fn="subtract_x",
+            constructor_name="create",
+            config={"x": 3},
+            eager=eager,
+            is_target=True,
+        ),
+    }
+
+    execution_context = ExecutionContext(graph_schema=graph_schema, model_id="1")
+
+    runner = DaskGraphRunner(
+        graph_schema=graph_schema, execution_context=execution_context
+    )
+    results = runner.run(inputs={"first_input": 3, "second_input": 4}, targets=["add"])
+    assert results == {"add": 7}
+
+
+@pytest.mark.parametrize("x, output", [(None, 5), (0, 5), (1, 4), (2, 3)])
+def test_default_config(x: Optional[int], output: int):
+    graph_schema: GraphSchema = {
+        "subtract": SchemaNode(
+            needs={"i": "input"},
+            uses=SubstractByX,
+            fn="subtract_x",
+            constructor_name="create",
+            config={"x": x} if x else {},
+            is_target=True,
+        ),
+    }
+
+    runner = DaskGraphRunner(
+        graph_schema=graph_schema,
+        execution_context=ExecutionContext(graph_schema=graph_schema, model_id="1"),
+    )
+    results = runner.run(inputs={"input": 5})
+    assert results["subtract"] == output
+
+
+def test_empty_schema():
+    runner = DaskGraphRunner(
+        graph_schema={},
+        execution_context=ExecutionContext(graph_schema={}, model_id="1"),
+    )
+    results = runner.run()
+    assert not results
+
+
+def test_no_inputs():
+    graph_schema: GraphSchema = {
+        "provide": SchemaNode(
+            needs={},
+            uses=ProvideX,
+            fn="provide",
+            constructor_name="create",
+            config={},
+            is_target=True,
+        ),
+    }
+    runner = DaskGraphRunner(
+        graph_schema=graph_schema,
+        execution_context=ExecutionContext(graph_schema=graph_schema, model_id="1"),
+    )
+    results = runner.run()
+    assert results["provide"] == 1
+
+
+def test_no_target():
```

Comment:
should we also test the case where we have an unused node in the graph (neither target nor parent of another node)?